confidential-data-hub/hub/src/hub.rs

```diff
@@ -43,7 +47,17 @@ impl DataHub for Hub {
         todo!()
     }
 
-    async fn get_resource(&self, _uri: String) -> Result<Vec<u8>> {
-        todo!()
+    async fn get_resource(&self, uri: String) -> Result<Vec<u8>> {
+        // to initialize a get_resource_provider client we do not need the ProviderSettings.
+        let mut client = kms::new_getter(&self.get_resource_provider, ProviderSettings::default())
```

Comment:
At first I uses kbcs and KMSes. When coding about secret unsealing and image decryption (UnwrapKey API), I found that the KBC's `get_resource()` semantics is totally same as what we want to define in vault:
```
async fn get_resource(&mut self, _rid: ResourceUri) -> Result<Vec<u8>>
async fn get_secret(&mut self, name: &str, annotations: &Annotations) -> Result<Vec<u8>>;
```

So if we put them together, it will be easy to do coding and higher level calls. Also, it has some extensibility:
- Potentially we can support the resource from a vault not only kbs, even if currently KBC/KBS is the most safe place.
- Let KBC/KBS to support vault-like secrets easily (They share the common APIs)